### PR TITLE
S4 ask questions

### PR DESCRIPTION
**Pull request by PapaPwn**
Just the basic ask question component. Tried to use https://github.com/froala/vue-froala-wysiwyg but couldn't figure out how 'webpack' works.

### CODE REVIEW

File: backend/src/models/question.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ class Question {
     postQuestion(courseID, { userID, title, body }) {
         return this.db
             .insert('question', { courseID, userID, title, body })
+            .then(() => this.getQuestions(courseID))
```
Comment (DarkPurple141) [inline backend/src/models/question.js]: Still don't think we should do it like this. :(. But it's increasingly baked into the codebase.

File: frontend/src/components/questions-answers/QuestionForm.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+</script>
+
+<style scoped lang='less'>
+
+.content {
+  padding: 0px 60px;
+}
+
+input[type=text] {
```
Comment (DarkPurple141) [inline frontend/src/components/questions-answers/QuestionForm.vue]: This should probably be a component.

File: frontend/src/views/Course.vue [hunk 3]
```diff
@@ -4,12 +4,18 @@
       title="Latest Questions"
       :questions="feed"
     />
+    <QuestionForm @submitQuestionForm="submitQuestion" class="questionForm">
+      <span class="form-failure"
+        v-if="error.code">{{error.message}}
+      </span>
+    </QuestionForm>
   </div>
 </template>
```
Comment (DarkPurple141) [inline frontend/src/views/Course.vue]: Some of this may clash with Nuno's branch.

**Comment by DarkPurple141** [comment]
Yes, render whenever no id prop is passed

File: frontend/src/router.js [hunk 4]
```diff
@@ -30,6 +30,14 @@ export default new Router({
       // which is lazy-loaded when the route is visited.
       component: () => import('./views/Course')
     },
+    {
+      path: '/question',
+      name: 'newQuestion',
+      props: ({params: { id }}) => ({
```
Comment (DarkPurple141) [inline frontend/src/router.js]: This won't work. There are no params and yet you're passing one in?

Comment (DarkPurple141) [reply frontend/src/router.js]: Unless you mean query.

**Comment by DarkPurple141** [comment]
@PapaPwn @NunoDasNeves There really needs to be a reset state function in the question store. IE you clear anything thats's in theory on some event.

**Comment by NunoDasNeves** [comment]
The code looks okay... but try going directly to a question. e.g. http://localhost:3000/question?qid=1

**Pull request merged**